src/pretty.rs

```diff
@@ -209,6 +248,18 @@ fn fmt(b: &mut String, limit: isize, mut width: isize, mut docs: Vector<(isize,
                     docs[0] = (indent, Mode::Broken, (*doc).clone());
                 }
             }
+
+            Document::FlexGroup(doc) => {
+                if !fits(limit - width, vector![(indent, Mode::Flex, (*doc).clone())]) {
+                    docs.push_front((
+                        indent,
+                        Mode::Flex,
+                        line().nest(2).append((*doc).clone()).append(line()),
```

Comment:
The document should not have hard coded indents in it, instead we should apply the nested when constructing the algebra.

With hard coded nesting this won't work for Erlang, which uses 4 space indentation, or for plain text, which doesn't indent at all.